api/v1alpha1/applicationsnapshot_types.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Images defines the various component images associated with the ApplicationSnapshot
+type Image struct {
+	// Component is the component name of the image
+	// +required
+	Component string `json:"component"`
+
+	// PullSpec is the particular version of the component's container image
+	// +required
+	PullSpec string `json:"pullSpec"`
+}
+
+// ApplicationSnapshotSpec defines the desired state of ApplicationSnapshot
+type ApplicationSnapshotSpec struct {
```

Comment:
@davidmogar do we want a reference to the Application in here too? I went back and forth in my head on it

Reply:
This is actually a good point. I think we still need to discuss about how these three CRDs (Application, ApplicationSnapshotEnvironmentBinding and ApplicationSnapshot) should be linked together. We went over it briefly in the past but nothing was agreed.

In my opinion having something like the following would make things easier:
* Snapshot CRD: List of images not linked to any Application.
* Bindings field in the Application CRD: It would be a list of named tuples:
```
bindings:
  - snapshot: ID
    environment: prod
```